test/test_distributions.py

```diff
@@ -1603,6 +1603,58 @@ def ref_log_prob(idx, x, log_prob):
 
         self._check_log_prob(Normal(loc, scale), ref_log_prob)
 
+    def test_normal_shape_cpu(self):
+        # CPU version normal is written in legacy code (TH).
+        # It won't support tensor broadcasting until it is migrated using new tensor iterator.
+        tensor23 = torch.rand(2, 3)
+        tensor21 = torch.rand(2, 1)
+        tensor6 = torch.rand(6)
+        tensor4 = torch.rand(4)
+        # normal case
+        self.assertEqual(torch.normal(tensor23, tensor23).size(), (2, 3))
+        # expandable (broadcasting) case (only when std or dev is scalar)
+        self.assertEqual(torch.normal(tensor23, 2).size(), (2, 3))
+        self.assertEqual(torch.normal(2, tensor23).size(), (2, 3))
+        # std and mean have same elements case
+        self.assertEqual(torch.normal(tensor23, tensor6).size(), (2, 3))
+        self.assertEqual(torch.normal(tensor6, tensor23).size(), (6,))
+        # std and mean have different number of element
+        with self.assertRaises(RuntimeError):
+            torch.normal(tensor23, tensor21)
+        with self.assertRaises(RuntimeError):
+            torch.normal(tensor23, tensor4)
+
+    @unittest.skipIf(not TEST_CUDA, "CUDA not found")
+    def test_normal_shape_gpu(self):
+        tensor12 = torch.rand(1, 2, device='cuda')
```

Comment:
(This isn't blocking) Is it possible to share code between `test_normal_shape_cpu` and `test_normal_shape_gpu`? 

On CPU, we support: (1) std/mean have same sizes, (2) std/mean have different sizes
On CUDA, we support: (1) std/mean have same sizes, (2) std/mean have different sizes, (3) std/mean are broadcastable

We could probably toss (1) and (2) into a helper function, but I'm not sure how much cleaner it would make this implementation. The reason why the tests are like this is because we implement CPU and CUDA separately...